node/src/components/network.rs

```diff
@@ -700,13 +700,30 @@ fn handle_identify_event(swarm: &mut Swarm<Behavior>, event: IdentifyEvent) {
 /// "/ip4/127.0.0.1/tcp/34553".
 fn address_str_to_multiaddr(address: &str) -> Multiaddr {
     let mut parts_itr = address.split(':');
-    let multiaddr_str = format!(
-        "/ip4/{}/tcp/{}",
-        parts_itr.next().expect("address should contain IP segment"),
-        parts_itr
-            .next()
-            .expect("address should contain port segment")
-    );
+    let multiaddr_str = if address
+        .chars()
+        .next()
+        .expect("cannot convert emptry address")
+        .is_numeric()
```

Comment:
This isn't robust: hostnames could begin with a digit.  I'm expecting this whole function to become obsolete once we remove the small_network component though and start using mutiaddr format in our config.  In which case, I'm ok with this for now.

Reply:
I've had the same thought a few hours after submitting this. I'm also in the "this should be gone soon" camp. Thanks!